packages/aggregator/src/connections/EntityConnection.js

```diff
@@ -13,6 +13,22 @@ export class EntityConnection extends ApiConnection {
   constructor(session) {
     const { getAuthHeader } = session;
     super({ getAuthHeader });
+
+    session.aggregatorEntityConnectionCache = session.aggregatorEntityConnectionCache || {};
```

Comment:
I rekon this is the part I'm most iffy about on this implementation, somehow just injecting an object into the session and then blindly using between instances feels wrong 😝 

Also, thoughts on the naming convention? I'd actually prefer if this `entityConnectionCache` could be shared between various `EntityConnections` for different parts of the system (eg. `report-server` and `aggregator`) and that's part of why I felt it'd be better just to cache the http request results (so that each client can then manipulate them however they see fit). If that were the case then the `entityConnectionCache` should really be an class which contains all the getCacheKey logic inside it. I shied away from the idea in the end since I didn't quite know where to put that class as it'd be used between packages (probably somewhere in server-boilerplate, `ApiConnection` maybe?)

Reply:
Yep I had a quick look through this on Friday and this was the main reason I decided not to fully review it and leave it for today. I'd definitely like to find a better place for it than on `session`, especially considering the context is passed in externally and given how static it is, there'd be no reason for an external client not to have that be much more long-lived than it currently is. I reckon `Aggregator` should have the cache stored somewhere that's entirely within its control, and then it can do its own cache invalidation based on whatever strategy ends up making sense. 

Is it too simplistic to have the cache just live on the `EntityConnection` instance, and then lift the instantiation of the entity connection up to happen at the same time the `Aggregator` class is created and subsequently be shared by all calls via that `aggregator`?

Reply:
We could also have a call about this when you're in tomorrow!